CompilerSource/compiler/compile.cpp

```diff
@@ -602,30 +604,25 @@ wto << "namespace enigma_user {\nstring shader_get_name(int i) {\n switch (i) {\
 
   idpr("Starting compile (This may take a while...)", 30);
 
-  string make = MAKE_flags;
+  string make = compilerInfo.make_vars["MAKEFLAGS"];
 
-  make += " Game ";
+  make += "Game ";
   make += "WORKDIR=\"" + eobjs_directory + "\" ";
   make += "CODEGEN=\"" + codegen_directory + "\" ";
-  make += mode == emode_debug? "GMODE=Debug ": mode == emode_design? "GMODE=Design ": mode == emode_compile?"GMODE=Compile ": "GMODE=Run ";
-  make += "GRAPHICS=" + extensions::targetAPI.graphicsSys + " ";
-  make += "AUDIO=" + extensions::targetAPI.audioSys + " ";
-  make += "COLLISION=" + extensions::targetAPI.collisionSys + " ";
-  make += "WIDGETS="  + extensions::targetAPI.widgetSys + " ";
-  make += "NETWORKING="  + extensions::targetAPI.networkSys + " ";
-  make += "PLATFORM=" + extensions::targetAPI.windowSys + " ";
-
-  if (CXX_override.length()) make += "CXX=" + CXX_override + " ";
-  if (CC_override.length()) make += "CC=" + CC_override + " ";
-  if (WINDRES_location.length()) make += "WINDRES=" + WINDRES_location + " ";
-
-  if (TOPLEVEL_cflags.length()) make += "CFLAGS=\"" + TOPLEVEL_cflags + "\" ";
-  if (TOPLEVEL_cppflags.length()) make += "CPPFLAGS=\"" + TOPLEVEL_cppflags + "\" ";
-  if (TOPLEVEL_cxxflags.length()) make += "CXXFLAGS=\"" + TOPLEVEL_cxxflags + "\" ";
-  if (TOPLEVEL_ldflags.length()) make += "LDFLAGS=\"" + TOPLEVEL_ldflags + "\" ";
-  if (TOPLEVEL_ldlibs.length()) make += "LDLIBS=\"" + TOPLEVEL_ldlibs + "\" ";
-
-  if (TOPLEVEL_rcflags.length()) make += "RCFLAGS=\"" + TOPLEVEL_rcflags + "\" ";
+  make += mode == emode_debug? "GMODE=\"Debug\"" : mode == emode_design? "GMODE=\"Design\"" : mode == emode_compile?"GMODE=\"Compile\"" : "GMODE=\"Run\"";
+  make += " ";
+  make += "GRAPHICS=\"" + extensions::targetAPI.graphicsSys + "\" ";
+  make += "AUDIO=\"" + extensions::targetAPI.audioSys + "\" ";
+  make += "COLLISION=\"" + extensions::targetAPI.collisionSys + "\" ";
+  make += "WIDGETS=\""  + extensions::targetAPI.widgetSys + "\" ";
+  make += "NETWORKING=\""  + extensions::targetAPI.networkSys + "\" ";
+  make += "PLATFORM=\"" + extensions::targetAPI.windowSys + "\" ";
+  make += "TARGET-PLATFORM=\"" + compilerInfo.target_platform + "\" ";
+  
+  for (const auto& key : compilerInfo.make_vars) {
+    if (key.second != "")
```

Comment:
This should probably be `if (!key.second.empty())` and not comparing to an empty string literal.